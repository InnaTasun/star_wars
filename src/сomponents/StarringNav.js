import ElemByUrl from './ElemByUrl';
import styled from 'styled-components';
import parseUrl from '../parseUrl';

function StarringNav(props) {
  const { dataArr, className } = props;
  let listItems = [];

  if (dataArr) {
    listItems = dataArr.map((filmUrl) => (
      <li className='starring-list--film' key={parseUrl(filmUrl).id}>
        <ElemByUrl url={filmUrl} mainProp='title' elemType='navImg' />
      </li>
    ));
  }
  return <ul className={`${className} starring-list`}>{listItems}</ul>;
}

const StyledStarringNav = styled(StarringNav)`
  display: flex;
  justify-content: start;
  flex-wrap: wrap;

  .starring-list {
    &--film {
      height: 290px;
      margin: 0.5rem;
      list-style-type: none;
      box-shadow: 7px 7px 7px rgba(0,0,0,0.5);
    }
  }
`;

export default StyledStarringNav;
