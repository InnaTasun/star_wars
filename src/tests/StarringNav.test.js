import React from 'react';
import { render, screen } from '@testing-library/react';
import StarringNav from '../сomponents/StarringNav';

const data = [
  'https://films/test/1/',
  'https://films/test/2/',
  'https://films/test/3/',
  'https://films/test/4/',
];

describe('StarringNav component', () => {
  test('renders with data', () => {
    render(<StarringNav dataArr={data} />);
    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.queryByRole('list').childElementCount).toBe(4);
  });

  test('renders without data', () => {
    render(<StarringNav />);
    expect(screen.getByRole('list').childElementCount).toBe(0);
  });

  test('renders with or without className', () => {
    let container = render(<StarringNav className='main-class' />).container;
    let firstDiv = container.firstElementChild;
    expect(firstDiv.className).toContain('main-class starring-list');

    container = render(<StarringNav className='test-class' />).container;
    firstDiv = container.firstElementChild;
    expect(firstDiv.className).toContain('test-class starring-list');

    container = render(<StarringNav />).container;
    firstDiv = container.firstElementChild;
    expect(firstDiv.className).not.toContain('main-class');
    expect(firstDiv.className).toContain('starring-list');
  });
});

test("StarringNav snapshot", () => {
  const { container } = render(<StarringNav />);
  expect(container).toMatchSnapshot();
});