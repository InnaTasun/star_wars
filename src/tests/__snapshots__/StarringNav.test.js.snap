// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarringNav snapshot 1`] = `
<div>
  <ul
    class="sc-gsDKAQ dsupdE starring-list"
  />
</div>
`;
